### ISSUE TITLE
Failed compile when use vue-tsc

### ISSUE THREAD

**Issue by amirameeen** [oruga-ui/oruga#509]
### Overview of the problem

**Oruga** version: 0.5.10
**Vuejs** version: 3.2.x
**OS/Browser**:

### Description

Failed compile the vue-tsc when I use oruga as reference/import.
Several issues come from missing types.
![image](https://user-images.githubusercontent.com/13668864/229732320-707b573b-b0f3-4df2-8ae7-793aa4b50c6e.png)

### Steps to reproduce

1. Install the oruga on your vue-ts and vite project.
2. Create a component and import one of the oruga component.
3. Generate the ts using vue-tsc

### Expected behavior

The vue-tsc can compile without error from oruga.

### Actual behavior

Can't finish the compile.

**Comment by jtommy** [comment]
@astagi did you fix it, right?

**Comment by Rtinox** [comment]
Hi,
I have the same issue

```
Errors  Files
    12  node_modules/@oruga-ui/oruga-next/src/components/input/Input.vue:12
     6  node_modules/@oruga-ui/oruga-next/src/components/icon/Icon.vue:98
     1  node_modules/@oruga-ui/oruga-next/src/components/input/index.ts:9
     4  node_modules/@oruga-ui/oruga-next/src/utils/BaseComponentMixin.ts:17
    27  node_modules/@oruga-ui/oruga-next/src/utils/FormElementMixin.ts:36
    11  node_modules/@oruga-ui/oruga-next/src/utils/helpers.ts:55
     1  node_modules/@oruga-ui/oruga-next/src/utils/useProgrammatic.ts:6
```

**Comment by mlmoravek** [comment]
@amirameeen should be fixed with v0.8.x

Feel free to reopen if you have any other typing problems.
